pkg/client/vault/client.go

```diff
@@ -205,3 +290,61 @@ func compressZCAP(zcap *zcapld.Capability) (string, error) {
 
 	return base64.URLEncoding.EncodeToString(compressed.Bytes()), nil
 }
+
+func lastElm(s, sep string) string {
+	all := strings.Split(s, sep)
+
+	return all[len(all)-1]
+}
+
+func buildEDVURI(h, vid, did string) string {
+	return fmt.Sprintf("%s/encrypted-data-vaults/%s/documents/%s", h, vid, did)
+}
+
+func prepareRemoteJWEKey(keyURL string, kmsImpl kms.KeyManager) ([]byte, string, error) {
+	kid := lastElm(keyURL, "/")
+
+	pubKeyBytes, err := kmsImpl.ExportPubKeyBytes(kid)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to retrieve main public key bytes from remote KMS: %w", err)
+	}
+
+	return pubKeyBytes, kid, nil
+}
+
+func encryptContent(wKMS KeyManager, wCrypto crypto.Crypto, content interface{}) (string, error) {
+	src,err := json.Marshal(content)
+	if err != nil {
+		return "", fmt.Errorf("marshal: %w", err)
+	}
+
+	_, kidURL, err := wKMS.Create(kms.NISTP256ECDHKW)
+	if err != nil {
+		return "", fmt.Errorf("create: %w", err)
+	}
+
+	pubKeyBytes, jweCryptoKID, err := prepareRemoteJWEKey(kidURL.(string), wKMS)
+	if err != nil {
+		return "", fmt.Errorf("prepare remote JWE key: %w", err)
+	}
+
+	ecPubKey := &crypto.PublicKey{KID: jweCryptoKID}
```

Comment:
you will need to populate all fields of ecPubKey, better yet unmarshall `pubKeyBytes` and use that instead because `kms.NISTP256ECDHKW`  type (and all of NISTP***ECDHKW and X25519 keys) is exported as `crypto.PublicKey`